File: POO - Python & Java/practica-1/src/gestorAplicacion/Factura.java
package gestorAplicacion;

import java.io.Serializable;

public class Factura  implements Serializable{
	
	/*	CLASS VARIABLES	*/
	private static final long serialVersionUID = 1L;
	private static final int NIT = 12;
	private static final String direccion = "Calle 1 #23";
	private static final String telefono = "#444 4444 (444)";
	
	/*	INSTANCE VARIABLES	*/
	private double valorPagar;
	
	/*	CONSTRUCTORS	*/
	public Factura(double valorPagar) {
		this(valorPagar, 0);
	}
	
	public Factura(double valorPagar, double descuento) {
		this.valorPagar = (1-descuento)*valorPagar;
	}
	
	/*	INSTANCE METHODS	*/
	public double devolucionDinero(Factura factura) {
		double valorDevolver = factura.getValorPagar();
		factura.setValorPagar(0);
		return valorDevolver;
	}
	
	/* GETTERS & SETTERS */
	public int getNIT() { return NIT; }
	
	public String getDireccion() { return direccion; }
	
	public String telefono() { return telefono; }
	
	public double getValorPagar() { return this.valorPagar; }
	public void setValorPagar(int valorPagar) { this.valorPagar = valorPagar; }
	
}

File: POO - Python & Java/practica-2/gestorAplicacion/Factura.py
class Factura:

    #	CLASS VARIABLES	
    _SERIALVERSIONUID = 1
    _NIT = 12
    _DIRECCION = "Calle 1 #23"
    _TELEFONO = "#444 4444 (444)"

    def __init__(self, valorPagar, descuento=0):
        self._valorPagar = (1-descuento)*valorPagar

    #	INSTANCE METHODS	
    def devolucionDinero(self, factura):
        valorDevolver = factura.getValorPagar()
        factura.setValorPagar(0)
        return valorDevolver

    # GETTERS & SETTERS 
    def getNIT(self):
        return Factura._NIT

    def getDireccion(self):
        return Factura._DIRECCION

    def telefono(self):
        return Factura._TELEFONO

    def getValorPagar(self):
        return self._valorPagar
    def setValorPagar(self, valorPagar):
        self._valorPagar = valorPagar

